Feature: Pythonization of ModuleNAO (#3642)
deepmodeling/abacus-develop: python/pyabacus/README.md

Installation
------------

> - Create and activate a new conda env, e.g. `conda create -n myenv python=3.8 & conda activate myenv`.

Reviewer: ```suggestion
- Create and activate a new conda env, e.g. `conda create -n pyabacus && conda activate pyabacus`.
```

Author: Since our package named as `pyabacus`, I was worried that user may feel confused and they can install other packages for DFT computation in their conda env.

Reviewer: ```suggestion
- Create and activate a new conda environment, e.g. `conda create -n myenv && conda activate myenv`.
```

- Clone ABACUS main repository and `cd abacus-develop/python/pyabacus`.
- Install dependencies by `pip install -r requirements.txt`.(Use `pip install -r requirements.txt -i https://pypi.tuna.tsinghua.edu.cn/simple` to accelerate installation process.)

> - Build pyabacus by `pip install -v .`.

Reviewer: Will `pip install -v .` automatically install the requirements?

Author: Yes, tested by @grysgreat. I think @grysgreat has followed the guide without installing `pybind` and `scikit-build-core` in your conda env manually?

Reviewer: > Yes, tested by @grysgreat. 

Then is the previous step not required?

> I think @grysgreat has followed the guide without installing `pybind` and `scikit-build-core` in your conda env manually?

Are the users required to install those packages? If so, please add them to requirements.txt .

Author: I see what you mean. `pip install -v.` will build the packages with building dependencies by reading the `.toml` file (e.g. pybind11, scikit-build) and the `pip install -r requirements.txt` will add the dependencies for usage or test call.

Reviewer 2: @caic99 I build pyabacus with pure gnu environment, which without installing pybind11 and scikit-build-core. And the pyabacus can properly build by following the Readme guide.

Author: @caic99 We can also replace the requirements.txt by adding such dependencies in `[project.optional-dependencies]` in the `.toml` file.

Reviewer: > @caic99 We can also replace the requirements.txt by adding such dependencies in `[project.optional-dependencies]` in the `.toml` file.

@jieli-matrix Looks good. A unified requirement list would be better.

Installation
------------

- Create and activate a new conda env, e.g. `conda create -n myenv python=3.8 & conda activate myenv`.
- Clone ABACUS main repository and `cd abacus-develop/python/pyabacus`.

> - Build pyabacus by `pip install -v .` or install test dependencies & build  pyabacus by `pip install .[test]`. (Use `pip install -r .[test] -i https://pypi.tuna.tsinghua.edu.cn/simple` to accelerate installation process.)

Reviewer: ```suggestion
- Build and install pyabacus by `pip install -v .`.
- To run tests, install test dependencies and build pyabacus by `pip install .[test]`.
```
